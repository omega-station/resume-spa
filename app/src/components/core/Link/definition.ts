export interface Props {
  /** Child elements */
  children: string | JSX.Element;
  /** URL */
  url: string;
  /** Target */
  target?: string;
  /** Title */
  title?: string;
}
