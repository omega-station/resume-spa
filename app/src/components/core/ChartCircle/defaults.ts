import { Props } from './definition';

const defaults: Props = {
  index: 0,
  label: '',
  percent: 100,
  duration: 1500,
};

export default defaults;
