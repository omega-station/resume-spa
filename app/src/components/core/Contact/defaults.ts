import { Props } from './definition';

const defaults: Props = {
  hasDescription: false,
  hasTitle: true,
  isPostalWithCity: false,
};

export default defaults;
