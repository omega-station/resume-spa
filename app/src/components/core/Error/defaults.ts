import { Props } from './definition';

const defaults: Props = {
  icon: 'fas-exclamation-triangle',
  message: 'Something went wrong',
};

export default defaults;
