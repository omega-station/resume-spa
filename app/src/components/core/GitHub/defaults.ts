import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  ariaLabel: 'View source on GitHub',
  color: color.black,
  fill: color.white,
  mode: 'square',
  url: process.env.REACT_APP_URL_GITHUB,
};

export default defaults;
