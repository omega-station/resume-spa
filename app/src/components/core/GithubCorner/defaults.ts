import { color } from '../../../utility/constant';
import { Props } from './definition';

const defaults: Props = {
  isLeft: false,
  isSmall: false,
  ariaLabel: 'View source on GitHub',
  color: color.core.white,
  fill: color.core.black,
  url: 'https://github.com/omega-station/resume-spa',
};

export default defaults;
