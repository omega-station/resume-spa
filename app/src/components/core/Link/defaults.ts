import { Props } from './definition';

const defaults: Props = {
  children: 'Default Link',
  url: '#',
  target: '_blank',
  title: '',
};

export default defaults;
