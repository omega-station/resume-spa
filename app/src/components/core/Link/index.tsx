import React from 'react';
import defaults from './defaults';
import { Props } from './definition';

const Link = (props: Props) => {
  const { children, target, title, url }: Props = { ...defaults, ...props };
  return (
    <a href={url} rel={target === '_blank' ? 'noopener noreferrer' : ''} target={target} title={title}>
      {children}
    </a>
  );
};

export default Link;
