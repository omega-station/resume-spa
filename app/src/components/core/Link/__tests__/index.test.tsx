import { MockedProvider } from '@apollo/react-testing';
import { act, cleanup, render } from '@testing-library/react';
import React from 'react';
import Link from '..';
import defaults from '../defaults';

afterEach(cleanup);

describe('Link', () => {
  it('should render with required props', () => {
    const component = render(<Link url={defaults.url}>{defaults.children}</Link>);
    expect(component).toMatchSnapshot();
  });

  it('should render with required props and target={default} prop', () => {
    const component = render(
      <Link url={defaults.url} target={defaults.target}>
        {defaults.children}
      </Link>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render with required props and title={default} prop', () => {
    const component = render(
      <Link url={defaults.url} title={defaults.title}>
        {defaults.children}
      </Link>
    );
    expect(component).toMatchSnapshot();
  });
});
