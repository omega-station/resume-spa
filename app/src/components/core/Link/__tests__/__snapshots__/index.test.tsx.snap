// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render with required props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="#"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        Default Link
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="#"
      rel="noopener noreferrer"
      target="_blank"
      title=""
    >
      Default Link
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Link should render with required props and target={default} prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="#"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        Default Link
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="#"
      rel="noopener noreferrer"
      target="_blank"
      title=""
    >
      Default Link
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Link should render with required props and title={default} prop 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        href="#"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        Default Link
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      href="#"
      rel="noopener noreferrer"
      target="_blank"
      title=""
    >
      Default Link
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
